stl/inc/experimental/random

```diff
@@ -0,0 +1,63 @@
+// random experimental header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _EXPERIMENTAL_RANDOM_
+#define _EXPERIMENTAL_RANDOM_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#include <random>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+namespace experimental {
+    inline namespace fundamentals_v3 {
+
+        _NODISCARD inline default_random_engine& _Engine() {
+            thread_local default_random_engine _Eng{random_device{}()};
+            return _Eng;
+        }
+
+        template <class _IntType>
+        _NODISCARD _IntType randint(const _IntType _Min, const _IntType _Max) {
+            _RNG_REQUIRE_INTTYPE(randint(), _IntType);
+            _STL_ASSERT(_Min <= _Max, "invalid min and max arguments for randint()");
+            return uniform_int_distribution<_IntType>{_Min, _Max}(_Engine());
+        }
+
+        inline void reseed() {
+            constexpr size_t _N = mt19937::state_size;
+            constexpr size_t _W = mt19937::word_size;
+            static_assert(_W % 32 == 0, "mt19937::word_size is not a multiple of 32");
+            constexpr size_t _K = _W / 32;
+
+            uint32_t _Values[_N * _K];
+            random_device _Rd;
+            _STD generate(_STD begin(_Values), _STD end(_Values), _STD ref(_Rd));
+            seed_seq _Seq{_STD cbegin(_Values), _STD cend(_Values)};
+
+            _Engine().seed(_Seq);
+        }

```

Comment:
We conventionally separate functions with newlines unless they are *highly* repetitive, which these overloads are not.